File: Contests/AtCoderPython/AtCoder Beginner Contest 262/PC.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr= new int[n];
        for(int i=0; i<n; i++){
            arr[i] = scanner.nextInt();
        }
        
        long match = 0;
        long ans = 0;
        for(int i = 0; i < n; i ++) {
            if(arr[i] == i + 1) {
                match ++;
            } else if(arr[i] > i + 1 && arr[arr[i]-1] == i + 1) {
                ans ++;
            }
        }
        if(match >= 2) {
            ans += match * (match - 1) / 2;
        }
        
        System.out.println(ans);
    }
}

File: Contests/AtCoderPython/AtCoder Beginner Contest 262/PC.py
from collections import Counter, defaultdict
from math import factorial
# from sortedcontainers import SortedSet, SortedList

N = int(input())
arr = list(map(int, input().split()))

match, ans = 0, 0
for i in range(N):
    if arr[i] == i + 1:
        match += 1
    elif arr[i] > i + 1 and arr[arr[i]-1] == i + 1:
        ans += 1

if match >= 2:
    # ans += int(factorial(match) / (2 * factorial(match - 2)))
    ans += int(match * ( match - 1 ) / 2)

print(ans)